tesseract_kinematics/include/tesseract_kinematics/ikfast/impl/ikfast_inv_kin.hpp

```diff
@@ -212,6 +236,29 @@ bool IKFastInvKin::checkInitialized() const
   return initialized_;
 }
 
+void IKFastInvKin::generateAllPossibleCombinations(const std::vector<std::vector<double>> free_joint_samples)
+{
+  free_joint_combos_.clear();
+  std::vector<std::size_t> curr_joint_indices(free_joint_samples.size(), 0);
+  while (curr_joint_indices.front() < free_joint_samples.front().size())
+  {
+    std::vector<double> curr_joint_values;
+    for (std::size_t i = 0; i < curr_joint_indices.size(); ++i)
+      curr_joint_values.push_back(free_joint_samples[i][curr_joint_indices[i]]);
+    free_joint_combos_.push_back(curr_joint_values);
+    curr_joint_indices.back()++;
+    for (std::size_t i = curr_joint_indices.size() - 1; i > 0; --i)
+    {
+      if (curr_joint_indices[i] == free_joint_samples[i].size())
+      {
+        curr_joint_indices[i] = 0;
+        curr_joint_indices[i - 1]++;
+      }
+    }
+  }
+  return;
+}
```

Comment:
I would make this a static function that takes the possible values for each joint as input and returns a vector of "free-joint states". Using this function to generate the input to the constructor seems a little more straightforward than initializing it with an empty set of free-joint states and overwriting that in a non-const way

Reply:
That's a much better way of doing it. Updated to match your suggestion.